tests/serverspec/default_spec.rb

```diff
@@ -0,0 +1,185 @@
+require "spec_helper"
+require "serverspec"
+require "nokogiri"
+
+package = "hadoop"
+extra_package = []
+
+service = "namenode"
+config_dir = "/etc/hadoop"
+user    = "hdfs"
+group   = "hadoop"
+ports   = [50_070]
+log_dir = "/var/log/hadoop"
+db_dir  = "/var/db/hadoop"
+default_user = "root"
+default_group = "root"
+
+case os[:family]
+when "freebsd"
+  package = "hadoop2"
+  extra_package = ["py27-lxml", "zookeeper"]
+  config_dir = "/usr/local/etc/hadoop"
+  default_group = "wheel"
+end
+
+describe package(package) do
+  it { should be_installed }
+end
+
+extra_package.each do |p|
+  describe package(p) do
+    it { should be_installed }
+  end
+end
+
+describe file("#{config_dir}/slaves") do
+  it { should exist }
+  it { should be_file }
+  it { should be_mode 644 }
+  its(:content) { should match(/^localhost$/) }
+end
+
+describe file(log_dir) do
+  it { should exist }
+  it { should be_mode 775 }
```

Comment:
`it { should be_directory }` should be here.
simple tests for files/directories should be done with a loop iterating an array of files/directories.